src/master-playlist-controller.js

```diff
@@ -612,6 +612,13 @@ export class MasterPlaylistController extends videojs.EventTarget {
       this.tech_.trigger('hls-reset');
     });
 
+    this.mainSegmentLoader_.on('segmenttimemapping', (event) => {
```

Comment:
I know we discussed it, but I'm still uneasy about using the event to communicate with videojs-contrib-media-sources. Although it's better not to add non-standard APIs to videojs-contrib-media-sources, making an event is essentially creating an API for it anyway, and makes the code harder to follow. Although it does help in cases where we don't want to fail on native media sources, I'm thinking that it might be better for us to have a specific method in source updater and contrib-media-sources to accept the data, just to make things easier to follow. Eventually contrib-media-sources will go away, removing the issue.

I do understand the arguments for using the event though (in that we aren't adding a "supported" non standard API to videojs-contrib-media-sources). If we do go with the event route, we should at least add some comments here and in videojs-contrib-media-sources to more easily understand how it is being passed between the two projects.

Reply:
I do understand that the event is more difficult to follow than a direct api, but since we plan on moving away from contrib-media-sources (sooner the better eh) and we already use this event approach else where, I added comments so we can save time on testing a refactor. I've also added comments in the contrib-media-sources pr